app/services/HashingService.coffee

```diff
@@ -0,0 +1,125 @@
+class HashingService
+
+  @reloadPageWithHash: (board, solutionPlacements, SolutionService) ->
+    unless @checkSolutionPlacements board, solutionPlacements, SolutionService
+      @emptyHash()
+      return false
+    HashingService.setHash board.initialValues, board.goals, solutionPlacements
+
+  @emptyHash: -> window.location.hash = ''
+
+  @setHash: (boardValues, goals, solutionPlacements) ->
+    window.location.hash = @encode boardValues, goals, solutionPlacements
+
+  @encode: (boardValues, goals, solutionPlacements) ->
+    boardValues = (JSON.stringify boardValues).replace(/(\[|\]|"|,|{|})*/g, '')
+
+    length = Math.sqrt boardValues.length
+    for list in [0...solutionPlacements.length]
+      for pos in [0...solutionPlacements[list].length]
+        solutionPlacements[list][pos] =
+          solutionPlacements[list][pos][0] * length +
+            solutionPlacements[list][pos][1]
+
+    btoa(JSON.stringify {b: boardValues, g: goals, p: solutionPlacements})
+
+  @decodeMap: () ->
+    try
+      decoded_s  = atob window.location.hash.substr(1, window.location.hash.length)
+      decoded = JSON.parse decoded_s
+    catch e
+      decoded = null
+    return decoded
+
+  @parse: (decoded) ->
+    return [[],[],[]] unless @successfulDecode decoded
+    length = Math.sqrt decoded.b.length
+    b = @decodeBoardValues decoded.b, length
+    g = @decodeGoals decoded.g
+    p = @decodeSolutionPlacements decoded.p, length
+    [b, g, p]
+
+  @successfulDecode: (decoded)->
+    return decoded? and
+           decoded.p? and
+           decoded.g? and
+           decoded.p? and
+           @regexPass decoded
+
+  @regexPass: (decoded) ->
+    alphabet = ['"', '{', '}', '[', ']', ',', ':',
```

Comment:
it could be better for us to put `alphabet` as a global constant for the hashing service so we don't have to reassign it each time we check